run_dir/static/js/data_del_plot.js

```diff
@@ -0,0 +1,357 @@
+            if (data[d][1].delivery_type == null){
+                series_name = "Other/undefined";
+            }else{
+                series_name = data[d][1].delivery_type;
+            }
+            view_color = view_coloring(series_name);
+        }else if (view_type == 'sample_type'){
+            if (data[d][1].sample_type.includes('Finished')){
+                series_name = "Finished Library";
+            }else if (data[d][1].sample_type.includes('total')){
+                series_name = "total RNA";
+            }else if (data[d][1].sample_type.includes('Tissue')){
+                series_name = "Tissue";
+            }else if (data[d][1].sample_type.includes('Genomic')){
+                series_name = "Genomic DNA";
+            }else if (data[d][1].sample_type.includes('Amplicon')){
+                series_name = "Amplicon";
+            }else{
+                series_name = "Other/undefined";
+            }
+            view_color = view_coloring(series_name);
+        }else if (view_type == 'best_practice_bioinformatics'){
+            if (data[d][1].best_practice_bioinformatics == 'Yes'){
+                series_name = "BP bioinformatics";
+            }else if (data[d][1].best_practice_bioinformatics == 'No'){
+                series_name = "No BP bioinformatics";
+            }else if (data[d][1].best_practice_bioinformatics == 'Special'){
+                series_name = "Special";
+            }else{
+                series_name = "Other/undefined";
+            }
+            view_color = view_coloring(series_name);
+            }
+        if (!series.hasOwnProperty(series_name)){
+            series[series_name] = {
+                name: series_name,
+                color: view_color,
+                data: [],
+            };
+            series.length += 1;
+        }
+        dp = {
+            y: data[d][1][key],
+            name: date_close,
+            pm: project_name,
+            ownURL: bioinfo_link
+        };
+        series[series_name].data.push(dp);
+        categories.push(date_close);
+        // Hackery to get a proper JS array for HCharts
+        var proper_series = []
+        for (s in series) {
+            proper_series.push(series[s]);
+        }
+    }
+  return [categories, proper_series];
+}
+
+function get_plot_data(key, name, view_type, search_string="", filter_inst_type){
+    $.getJSON('/api/v1/proj_staged/'+search_string, function(data) {
+        window.current_plot_data=data;
+        update_color_schemes(data);
+        make_plot(key, name, view_type, filter_inst_type);
+    });
+}
+//Since we have been confused by the coloring in the past, I set a few default colors to re-use for simplicity.
+function view_coloring(series_name){
+    switch(series_name){
+        case "NovaSeq":
+        case "HDD":
+        case "RNA-Seq":
+        case "Finished Library":
+        case "BP bioinformatics":
+            return current_color_schemes[0].hex();
```

Comment:
Why not just return the `chroma` colour directly?
```suggestion
            return chroma('hotpink').hex();
```

Reply:
That is true, originally I thought we might need a function to change the colors dynamically as in the flowcell trends plot, but this is so much easier to follow. Yes, correct, nice catch, much cleaner!

Reply:
![image](https://user-images.githubusercontent.com/465550/109050006-3ab73c80-76d9-11eb-8cc8-cad88b630c16.png)

Better to just commit PR suggestions like this via the GitHub PR interface if you agree with them ☝🏻 

See https://docs.github.com/en/github/collaborating-with-issues-and-pull-requests/incorporating-feedback-in-your-pull-request

Reply:
Note that the docs don't say it, but best to do it in a batch using the _"Add suggestion to batch"_ button instead of having a commit for each thing. This button only works when you're looking at the _Files Changed_ tab, which is super unintuitive.

Reply:
OK good to know!